Lib/test/datetimetester.py

```diff
@@ -2676,6 +2676,17 @@ def test_fromisoformat_fails_datetime(self):
                 with self.assertRaises(ValueError):
                     self.theclass.fromisoformat(bad_str)
 
+    def test_fromisoformat_fails_surrogate(self):
+        # Test that when fromisoformat() fails with a surrogate character as
+        # the separator, the error message contains the original string
+        dtstr = "2018-01-03\ud80001:0113"
+
+        with self.assertRaises(ValueError) as cm:
+            self.theclass.fromisoformat(dtstr)
+
+        msg = cm.exception.args[0]
```

Comment:
I was having trouble using a `assertRaisesRegex` here, I think because of the escape characters, but `in` works fine.

Reply:
You could use `assertRaisesRegex(ValueError, re.escape(f"{dtstr!r}"))`.

Reply:
Ah, thank you! I thought there must be a way to do this but it seemed like a lot of work to figure it out. Thanks for the pointer. :smile: